### PR TITLE
[core,connection] handle message channel in state

### PR DESCRIPTION
**Pull request by akallabeth**
When in demand active it is possible that we can receive message channel messages. In case we receive one switch parsing to the appropriate handlers.

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/core/connection.c [hunk 1]
```diff
@@ -1264,6 +1264,21 @@ state_run_t rdp_client_connect_demand_active(rdpRdp* rdp, wStream* s)
 	if (freerdp_shall_disconnect_context(rdp->context))
 		return STATE_RUN_QUIT_SESSION;
 
+	if (rdp->mcs->messageChannelId && (channelId == rdp->mcs->messageChannelId))
```
Comment (hardening) [inline libfreerdp/core/connection.c:1267]: Looks ok for me, but don't we need to mutualize that, as I guess that we might get some inflight packets in other states, and also get some different packets ?

Comment (akallabeth) [reply libfreerdp/core/connection.c:1267]: @hardening good point. waiting for feedback, and if yes wrap this into a common function.

**Comment by jens-maus** [comment]
@akallabeth I checked the sources of this PR and now a connection redirection is possible against our connection broker. So yes, this PR seems to solve the issues we were having with xfreerdp3 lately. Good job!

File: libfreerdp/core/connection.c (hunk 1 above)
Comment (akallabeth) [reply libfreerdp/core/connection.c:1294]: ok, done, refactored this where possible.

**Comment by akallabeth** [comment]
@freerdp-bot test please

**Pull request merged**